crates/bevy_render/src/view/visibility/render_layers.rs

```diff
@@ -20,9 +21,15 @@ pub type Layer = u8;
 /// An entity with this component without any layers is invisible.
 ///
 /// Entities without this component belong to layer `0`.
-#[derive(Component, Copy, Clone, Reflect, PartialEq, Eq, PartialOrd, Ord)]
+#[derive(Component, Clone, Reflect, PartialEq, Eq, PartialOrd, Ord)]
 #[reflect(Component, Default, PartialEq)]
-pub struct RenderLayers(LayerMask);
+pub struct RenderLayers(SmallVec<[u64; 1]>);
```

Comment:
```suggestion
pub struct RenderLayers(SmallVec<[u64; 2]>);
```

With the `union` feature on, we can fit 2 u64s on the stack before it starts to grow bigger than a base `Vec` (on 64-bit platforms), alongside the one usize for the capacity check. This should support at least 128 layers before it spills onto the heap.

Reply:
Thanks, this is a good call out on this feature.